frappe/model/db_query.py

```diff
@@ -328,7 +328,38 @@ def prepare_filter_condition(self, f):
 		can_be_null = True
 
 		# prepare in condition
-		if f.operator.lower() in ('in', 'not in'):
+		if f.operator.lower() in ('ancestors of', 'descendants of'):
+			values = f.value or ''
+
+			# TODO: handle list and tuple
+			# if not isinstance(values, (list, tuple)):
+			# 	values = values.split(",")
+
+			ref_doctype = f.doctype
+
+			if frappe.get_meta(f.doctype).get_field(f.fieldname) is not None :
+				ref_doctype = frappe.get_meta(f.doctype).get_field(f.fieldname).options
+
+			result=[]
+			lft, rgt = frappe.db.get_value(ref_doctype, f.value, ["lft", "rgt"])
+
+			# Get descendants elements of a DocType with a tree structure
+			if f.operator.lower() in ('descendants of') :
```

Comment:
how about additional filters rather than `in` ?

Reply:
added not filters